ipam/ipv6Ipam_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Microsoft. All rights reserved.
+// MIT License
+
+package ipam
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/common"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	testclient "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	testNodeName   = "TestNode"
+	testSubnetSize = "/126"
+)
+
+func newKubernetesTestClient() kubernetes.Interface {
+
+	nodeName := testNodeName
+	testnode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+		},
+		Spec: v1.NodeSpec{
+			PodCIDR:  "10.0.0.1/24",
+			PodCIDRs: []string{"10.0.0.1/24", "ace:cab:deca:deed::/64"},
+		},
+	}
+
+	client := testclient.NewSimpleClientset()
+	client.CoreV1().Nodes().Create(context.TODO(), testnode, metav1.CreateOptions{})
```

Comment:
will this test work in non-kubernetes environment?

Reply:
testclient comes from k8s.io/client-go/kubernetes/fake, which gives a fake clientset used for tests